app/src/main/java/com/kleshchin/danil/filemanager/MainActivity.java

```diff
@@ -0,0 +1,137 @@
+package com.kleshchin.danil.filemanager;
+
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.PorterDuff;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Environment;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.ActionBar;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.MenuItem;
+import android.view.View;
+import android.webkit.MimeTypeMap;
+import android.widget.AdapterView;
+import android.widget.EditText;
+import android.widget.ListView;
+import android.widget.Toast;
+import java.io.File;
+import static android.content.Intent.FLAG_ACTIVITY_NEW_TASK;
+
+public class MainActivity extends AppCompatActivity {
+
+    //    private static final String MAIN_PATH = "/storage";
+    private static final String MAIN_PATH = Environment.getExternalStorageDirectory().getPath();
+    private static final String LIST_VIEW_STATE = "listview.state";
+
+    private ListView listView_;
+    private ListAdapter listAdapter_;
+    private EditText toolbarTitle_;
+    private File currentShowingPath_;
+
+    private static void onFileSelected(String path, @NonNull Context context)
```

Comment:
Also when you know that the method can throw an exception this should be declared explicitly.
https://docs.oracle.com/javase/tutorial/essential/exceptions/declaring.html